### PR TITLE
fix failed graph x axis

### PR DESCRIPTION
**Pull request by DanielMwendwa**
Closes: #180 
Update failed msg graph to have one day view and 10 minutes view with a y axis control button

### CODE REVIEW

File: public/graph.js [hunk 1]
```diff
@@ -730,15 +718,181 @@ class GraphController {
...
+            if (isYLimitFailedManuallySet != true) {
+                yLimitFailed = yLimitFailedTotal;
+            }
+
+            // set scale domain for failed graph
+            y_total_failed_sms.domain([0, yLimitFailed]);
+            failed_messages_x_axis_range.domain(d3.extent(data, d => new Date(d.datetime)));
+
+            d3.selectAll(".redrawElementFailed").remove();
+            d3.selectAll("#failedLine10min").remove();
```
Comment (IsaackMwenda) [inline public/graph.js]:
```suggestion
            d3.selectAll("#failedLine10Min").remove();
```

File: public/graph.js [hunk 2]
```diff
@@ -730,15 +718,181 @@ class GraphController {
...
+            d3.selectAll("#failedLine1day").remove();
```
Comment (IsaackMwenda) [inline public/graph.js]:
```suggestion
            d3.selectAll("#failedLine1Day").remove();
```

File: public/graph.js [hunk 3]
```diff
@@ -730,15 +718,181 @@ class GraphController {
...
+                .append("text")
+                .attr("class", "redrawElementFailed")
+                .attr(
+                    "transform",
+                    "translate(" + Width / 2 + " ," + (Height + Margin.top + 50) + ")"
+                )
+                .style("text-anchor", "middle")
+                .text("Time (D:H:M:S)");
+            
+            // Add the Y Axis for the total failed sms graph
```
Comment (IsaackMwenda) [inline public/graph.js]: The order for the other graphs is add Y axis then X axis rearrange this for code readability

File: public/graph.js [hunk 4]
```diff
@@ -730,15 +718,181 @@ class GraphController {
...
+                .attr("y", 0 - Margin.top / 2)
+                .attr("text-anchor", "middle")
+                .style("font-size", "20px")
+                .style("text-decoration", "bold")
+                .text("Total Failed Messages(s) / day");
+
+            // Label Lines for the total failed sms graph
+            total_failed_sms_graph.append("text");
+
+            // Define line paths for total failed sms(s)
```
Comment (IsaackMwenda) [inline public/graph.js]: Let draw a bar chart instead of a line chart as is on the other two graphs.

File: public/graph.js [hunk 5]
```diff
@@ -730,15 +718,181 @@ class GraphController {
...
+            // Add X axis label for the total failed sms graph
+            total_failed_sms_graph
+                .append("text")
+                .attr("class", "redrawElementFailed")
+                .attr(
+                    "transform",
+                    "translate(" + Width / 2 + " ," + (Height + Margin.top + 50) + ")"
+                )
+                .style("text-anchor", "middle")
+                .text("Time (D:H:M:S)");
```
Comment (IsaackMwenda) [inline public/graph.js]:
```suggestion
                .text("Time (D-M-Y)");
```

**Review by IsaackMwenda — CHANGES_REQUESTED** [review]

File: public/graph.js (hunk 3 above)
Comment (DanielMwendwa) [reply public/graph.js]: Let me change when I'll be creating bar graphs

File: public/graph.js [hunk 6]
```diff
@@ -730,15 +718,169 @@ class GraphController {
...
+            y_total_failed_sms.domain([0, yLimitFailed]);
+
+            d3.selectAll(".redrawElementFailed").remove();
+            d3.selectAll("#failedBarChart").remove();
+            d3.selectAll("#failedBarChart10min").remove();
+
+            // Add the Y Axis for the total failed sms graph
+            total_failed_sms_graph
+                .append("g")
+                .attr("id", "axisSteelBlue")
```
Comment (IsaackMwenda) [inline public/graph.js]: Lets use a more representative color #a82e2e ?

**Review by IsaackMwenda — APPROVED** [review]
Thanks

**Pull request merged**